routes/Routes.js

```diff
@@ -0,0 +1,159 @@
+import express, { response } from "express";
+import listEndpoints from "express-list-endpoints";
+import HappyThought from "../model/Happythoughts";
+const router = express.Router();
+
+// get 20 happy thoughts
+router.get("/thoughts", async (req, res) => {
+  try {
+    const happyThoughts = await HappyThought.find()
+      .sort({ createdAt: "desc" })
+      .limit(20)
+      .exec();
+    res.json(happyThoughts);
+  } catch (error) {
+    res.status(400).json({
+      success: false,
+      response: error,
+      message: "Could not get thoughts",
+    });
+  }
+});
+
+//post new happy thought
+router.post("/thoughts", async (req, res) => {
+  const { message } = req.body;
+  try {
+    const newHappyThought = await new HappyThought({ message }).save();
+    res.status(201).json(newHappyThought);
+  } catch (error) {
+    console.error(error); // Log the error
+    res.status(400).json({
+      success: false,
+      response: error,
+      message: "Could not save thought",
+    });
+  }
+});
+
+//post a like to a happy thought
+router.post("/thoughts/:thoughtId/like", async (req, res) => {
+  const { thoughtId } = req.params;
+  try {
+    const updatedThought = await HappyThought.findByIdAndUpdate(
+      thoughtId,
+      { $inc: { hearts: 1 } },
+      { new: true }
+    );
+    if (updatedThought) {
+      res.json(updatedThought);
+    } else {
+      res.status(404).json({
+        success: false,
+        response: error,
+        message: "Thought not found",
+      });
+    }
+  } catch (error) {
+    res.status(400).json({
+      success: false,
+      response: error,
+      message: "Could not save like",
+    });
+  }
+});
+
+//delete a happy thought
+router.delete("/thoughts/:thoughtId", async (req, res) => {
+  const { thoughtId } = req.params;
+  try {
+    const deletedThought = await HappyThought.findByIdAndDelete(thoughtId);
+    if (deletedThought) {
+      res.json(deletedThought);
+    } else {
+      res.status(404).json({
+        success: false,
+        response: error,
+        message: "Thought not found",
+      });
+    }
+  } catch (error) {
+    res.status(400).json({
+      success: false,
+      response: error,
+      message: "Could not delete thought",
+    });
+  }
+});
+
+//filter happy thoughts by limit or skip or both
+router.get("/thoughts/q", async (req, res) => {
+  const { limit, page } = req.query;
+  const thoughts = await HappyThought.find()
+    .sort({ createdAt: "desc" })
+    .limit(parseInt(limit))
+    .skip(parseInt(page))
+    .exec();
+  res.json(thoughts);
+});
+
+// get updated documentation
+router.get("/", (req, res) => {
```

Comment:
Most generic endpoints should always come first, e.g.

1. /
2. /books
3. /books/:id